File: hw/villagomez/u2/hw18ZooJS/ec.edu.espe.zoo.model/Fish.js
const Animal = require('./Animal');
const Cage = require('./Cage'); 

class Fish extends Animal {

    constructor(id, description, dateOnborn, gender, numberOfBones, cage) {
        if (new.target === Fish) {
            throw new Error('Cannot instantiate abstract class Fish');
        }
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
}

module.exports = Fish;

File: ws/rodriguez/u2/ws24/Zoo/Zoo/src/main/java/ec/edu/espe/zoo/model/Fish.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author David Rodriguez,THEJAVABANDITS DCCO-ESPE
 */
public abstract class Fish extends Animal{
    
    public Fish(int id, String description, Date dateOnborn, char gender, int numberOfBones, Cage cage) {
        super(id, description, dateOnborn, gender, numberOfBones, cage);
    }
    
}
